proposed/0013-immutable-data-naming/0013-immutable-data-naming.md

```diff
@@ -13,75 +13,91 @@ This RFC outlines the system components and design for how the three immutable d
 
-# Alternatives
+When receiving a successful Get response for a Backup or Sacrificial chunk, if this is to be sent as a response to a client it must be converted into a Normal chunk.
 
-1. The SAFE network itself is able to be free of carrying out any naming calculation and handling based on types, as long as client be aware of such and fire requests bearing the different type-dependent name.  This will have the least impact to the current code base, however the client app must need to be aware of that and carry out its duty. It also leaves an option (probably good) when the client app decides only one of the type will be enough.
+It is expected that three groups should never all fail for any given chunk.
 
+### Churn
 
-# Unresolved Questions
+If a current PN ceases to be a holder for a chunk due to churn (either it has disconnected or it has been pushed out of the close group for that chunk due to other Vaults joining the network) then it should be removed from the list of holders and a replacement copy should be Put to a new PN.  This will likely involve following the Get procedure first.
 
-1. It is notionable that to reduce the code coplexity, the portal DataManager group of an immutable data shall still be DM(normal_name). Then such group can forwarding the requests to pmid_nodes closest to normal_name, backup_name, sacrificial_name.
-However, due to the facts the pmid_nodes all already closest nodes to the data_manager, such forwarding mechanism may not be able to securse a highly diversity of the distribution. i.e. some pmid_nodes may holding different type copies at the same time.
+## Planned Work
 
-2. Fraser argued that the hash of the hash paradigm shall not be used, since this could yield the same group for the normal, backup and sacrificial copies (the probability of this increasing to 1 as the network size decreases).  He proposed to XOR the chunk ID ( hash(content) ) to yield each group. While this should be more efficient than repeated hashing, more importantly it should yield groups which don't intersect (except for tiny networks).
-Essentially the "normal" group will be XOR with 000...000 i.e. equivalent to the ID.
-The sacrificial group will be XOR with fff...fff, i.e. the furthest location in the address space. And the backup will be in the middle; i.e. XOR with 800...000.
+1. MaidManager
+    1. Block attempts to Put or Get Backup or Sacrificial chunks
 
+1. DataManager
+    1. Put Backup and Sacrificial copies when handling a client Put
+    1. Handle Puts from other DMs for Backup and Sacrificial chunks
+    1. Handle failure to Get by trying to Get a different type of the same chunk
+    1. Handle successful Get response of non-Normal type where this needs converted to a Normal chunk to satisfy a client Get request
+    1. Handle failure to Put on PN by Deleting Sacrificial chunks on that Vault and retrying Put
```

Comment:
this involves a iterate through all the account entry to figure out which sacrificial copies are stored on that PN.